src/common/res/features/budget-progress-bars/main.js

```diff
@@ -40,46 +53,69 @@
       }
 
       function addGoalProgress(subCategoryName, target) {
-        if ( "Uncategorized Transactions" != subCategoryName ) {
+        if ( 'Uncategorized Transactions' != subCategoryName ) {
           calculation = getCalculation(subCategoryName);
 
-          var status = 0;
-          var hasGoal = false;
-          if (calculation.goalExpectedCompletion > 0 || calculation.goalOverallLeft) {
-            // Target total goal
-            hasGoal = true;
-            status = calculation.balance / (calculation.balance + calculation.goalOverallLeft);
-          }
-          else if (calculation.goalTarget > 0) {
-            // Taget by date
-            // or Montly goal
-            hasGoal = true;
-            status = 1 - calculation.goalUnderFunded / calculation.goalTarget;
-          }
-          else if (calculation.upcomingTransactions < 0) {
-            // Upcoming transactions "goal"
-            hasGoal = true;
-            status = - calculation.balance / calculation.upcomingTransactions;
-          }
+					var status = 0;
+					var tstatus = 0;
+					var hasGoal = false;
 
-          if (hasGoal) {
-            status = status > 1 ? 1 : status;
-            status = status < 0 ? 0 : status;
-            var percent = Math.round(parseFloat(status)*100);
-            target.style.background = "linear-gradient(to right, rgba(22, 163, 54, 0.3) " + percent + "%, white " + percent+ "%)";
-          }
-          else {
-            target.removeAttribute("style");
-          }
-        }
-      }
+					switch (calculation.goalType) {
+						case 'TB' :
+							hasGoal = true;
+
+							if ( calculation.balance >= calculation.targetBalance ) {
+								status = 100;
+							} else {
+								status = calculation.balance / (calculation.balance + calculation.goalOverallLeft);
+							}
+
+							break;
+						case 'TBD' :
```

Comment:
It probably could be combined. I'll take a boo tonight.

On Thu, Mar 31, 2016 at 6:37 AM, Kevin Brown <notifications@github.com>
wrote:

> In src/common/res/features/budget-progress-bars/main.js
> <https://github.com/toolkit-for-ynab/toolkit-for-ynab/pull/269#discussion_r58039349>
> :
>
> > -            target.removeAttribute("style");
> > -          }
> > -        }
> > -      }
> > +					switch (calculation.goalType) {
> > +						case 'TB' :
> > +							hasGoal = true;
> > +
> > +							if ( calculation.balance >= calculation.targetBalance ) {
> > +								status = 100;
> > +							} else {
> > +								status = calculation.balance / (calculation.balance + calculation.goalOverallLeft);
> > +							}
> > +
> > +							break;
> > +						case 'TBD' :
>
> Could this case be combined with the code above? So:
>
> case 'TB' :case 'TBD' :
>     hasGoal = true;
>
>     if ( calculation.balance >= calculation.targetBalance ) {
>         status = 100;
>     } else {
>         status = calculation.balance / (calculation.balance + calculation.goalOverallLeft);
>     }
>
>     break;// Other cases
>
> Looks like the same code to me.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/toolkit-for-ynab/toolkit-for-ynab/pull/269/files/966ca7599f9cb51d9f2414bda5129d6aa5781ae7#r58039349>
>